app/middleware/cache_headers.py

```diff
@@ -44,10 +45,11 @@ def __call__(self, request):
             patch_response_headers(response, cache_timeout=10)
         elif (
             request.method in ('GET', 'HEAD') and
-            not request.path.startswith(urlparse(settings.STATIC_URL).path)
+            not request.path.startswith(urlparse(settings.STATIC_URL).path) and
+            get_max_age(response) is None  # only set if not already set by the application
```

Comment:
So we include 40X intentionally here, right?
Because they should not only be accidentally / temporarily returned, so they can be cached. Do I understand that correctly?

Reply:
40x errors are cached the same way as 20x calls, 40x are error due to client wrong usage opposed to 50x which is a server error that might be temporar (e.g. 502 or 504 due to server overloaded).
Here in the line 48 and 49 we don't overwrite cache settings if the view already set it.